### PR TITLE
LPS-189359 Unsatisfied reference at com.liferay.portal.monitoring on Wildfly, Websphere, and JBoss

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-189359

### CODE REVIEW

**Comment by MrJohn1911** [comment]
Hey @dantewang, fyi I didn't understand why the innerclass strategy caused this specific unsatisfied reference in these servers, so I just extracted the service from the original class and created a new component, but I don't know if this is the best fix for it, do you have another idea?

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by catarineaguiar** [comment]
no regressions found. 

The blocker was fixed with these changes.

**Review by RenanUmV — APPROVED** [review]

**Comment by dantewang** [comment]
@MrJohn1911 The reason is that the `MonitoringFilter`, like many other services, is lazy. Only when it's tracked, it will be activated. From my debugging on Wildfly, the `MonitoringFilter.activate` is not invoked until `InvokerFilterHelper.init`, which is triggered by my first attempt to open `localhost:8080` in browser.

So, the `MonitoringFilter` is not activated during startup, resulting in the `PortalMonitoringControl` not being registered.

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/servlet/filter/PortalMonitoringControlImpl.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+package com.liferay.portal.monitoring.internal.servlet.filter;
+
+import com.liferay.portal.kernel.monitoring.PortalMonitoringControl;
+
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * @author Joao Victor Alves
+ */
+@Component(service = PortalMonitoringControl.class)
```
Comment (dantewang) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/servlet/filter/PortalMonitoringControlImpl.java:24]: @MrJohn1911 Can you make this one `enabled=false` and see if it works? Basically all the components in this module should be disabled by default.

File: modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/servlet/filter/PortalMonitoringControlImpl.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+public class PortalMonitoringControlImpl implements PortalMonitoringControl {
+
+	@Override
+	public boolean isMonitorPortalRequest() {
+		return MonitorPortalRequestManagerUtil.isMonitorPortalRequest();
```
Comment (dantewang) [inline modules/apps/portal/portal-monitoring/src/main/java/com/liferay/portal/monitoring/internal/servlet/filter/PortalMonitoringControlImpl.java:29]: @MrJohn1911 Can you inline everything of the util into this component, and let `MonitoringFilter` reference this component directly?

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
Please see my comment and resend.

**Pull request closed without merging**